### ISSUE TITLE
BackgroundTask's "Run now" action does not respond

### ISSUE THREAD

**Issue by Xteuk** [M-Files/VAF.Extensions.Community#136]
Hello,

Since I've updated to last version (from a very old one), I'm unable to make the "Run now" action to work.
I've created the simplest VAF to try this, and I've tried plenty arguments of "TaskQueueBackgroundOperation.DashBoardRunCommand".
The link is visible in the Dashboard, its text can be changed, but still nothing's triggered.

It's still possible I'm using it incorrectly, of course. Can you help, please ?

On StartApplication() :

![2024-10-22_13h51_35](https://github.com/user-attachments/assets/e033ec0b-2cd6-4c25-9015-7c6e24edc405)

On OnConfigurationUpdated()

![image](https://github.com/user-attachments/assets/34a9bc7b-f39a-45b9-883b-13e65b2a7be0)

[MFiles.Fealinx.TestVAF.zip](https://github.com/user-attachments/files/17475614/MFiles.Fealinx.TestVAF.zip)

**Comment by CraigHawker** [comment]
The TaskBackgroundOperationManager was designed as a "sticking plaster" over the old background operations.  We would recommend migrating to the attribute-based approach instead: https://github.com/M-Files/VAF.Extensions.Community/tree/master/MFiles.VAF.Extensions#using-vaf-23-task-processors

**Comment by CraigHawker** [comment]
I just debugged the code and it appears that there is some issue returning the commands for the background operations manager.  For now I suggest two alternates:

1. Migrate to the new approach (recommended)
2. Alter your code as below

```csharp
// In VaultApplication.cs:
public override IEnumerable<CustomDomainCommand> GetCommands(IConfigurationRequestContext context)
{
	var commands = base.GetCommands(context).ToList();
	if(!commands.Contains(this.SyncBackgroundOperation.DashboardRunCommand))
		commands.Add(this.SyncBackgroundOperation.DashboardRunCommand); // Add your command.
	return commands;
}
```

**Comment by CraigHawker** [comment]
I'm going to re-open this as I am going to take a quick look to see whether I can resolve this now.

In general the task processor approach should be able to be just as configurable as the other one, but the infrastructure is so much easier to get going with.  You may, yes, end up with a smaller number of queues and task types (because those have to be compile-time-constants with the attribute approach), but that's not necessarily a practical constraint.

**Comment by CraigHawker** [comment]
Should be resolved in version `24.10.72`, currently being pushed to nuget.org.
